tasking/pulp/tasking/scheduler.py

```diff
@@ -117,38 +114,39 @@ def check_celery_processes(self):
         msg = _('Checking if pulp_workers, pulp_celerybeat, or pulp_resource_manager processes '
                 'are missing for more than %d seconds') % constants.CELERY_TIMEOUT_SECONDS
         _logger.debug(msg)
-        now = ensure_tz(datetime.utcnow())
+        now = timezone.now()
         oldest_heartbeat_time = now - timedelta(seconds=constants.CELERY_TIMEOUT_SECONDS)
-        worker_list = Worker.objects.all()
-        worker_count = 0
-        resource_manager_count = 0
-        scheduler_count = 0
-
-        for worker in worker_list:
-            if worker.last_heartbeat < oldest_heartbeat_time:
-                msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
-                _logger.error(msg)
-
-                if worker.name.startswith(constants.SCHEDULER_WORKER_NAME):
-                    worker.delete()
-                else:
-                    delete_worker(worker.name)
-            elif worker.name.startswith(constants.SCHEDULER_WORKER_NAME):
-                scheduler_count = scheduler_count + 1
-            elif worker.name.startswith(TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME):
-                resource_manager_count = resource_manager_count + 1
+
+        for worker in Worker.objects.filter(last_heartbeat__lt=oldest_heartbeat_time):
+            msg = _("Worker '%s' has gone missing, removing from list of workers") % worker.name
+            _logger.error(msg)
+
+            if worker.name.startswith(constants.SCHEDULER_WORKER_NAME):
+                worker.delete()
             else:
-                worker_count = worker_count + 1
+                _delete_worker(worker.name)
 
-        if resource_manager_count == 0:
+        if not (Worker.objects.filter(name__startswith=constants.RESOURCE_MANAGER_WORKER_NAME)
```

Comment:
I rearranged things and now just use the count variable that used to be declared just after this code.